Point `node_modules/bs-platform` to `melange` to disguise `melange` as `bsb`

bhootd opened the pull request:
Would you be willing to merge this change? I would rather have this in the original template instead of maintaining a fork.

Commit:
set melange to be used as bsb

Commit:
update README.md to reflect linking of melange as bsb

jchavarri commented on .gitignore:
```diff
@@ -6,4 +6,6 @@ _esy
-dune.bsb
\ No newline at end of file
+dune.bsb
+esy.lock
```
I don't think we want to ignore `esy.lock` folder, checking this folder in git makes sure everyone has the same versions of everything.

bhootd replied:
Wasn't aware of this. I have pushed a commit to this effect.

jchavarri commented on README.md:
```diff
@@ -20,10 +20,15 @@ Then:
-to build the project. Now you should see a `_build` folder with all generated files, you can run
+to build the project. This command, as constructed in `esy.json`,
+
+1. generates a symbolically linked dir `bs-platform` in `node_modules`, which effectively makes `melange` the compiler behind the old `bsb` command, and then
```
```suggestion
1. generates a symbolically linked dir `bs-platform` in `node_modules`, which is helpful when using JavaScript bundlers like Webpack that expect to find Melange stdlib JavaScript files under `node_modules/bs-platform`.
```

jchavarri commented on README.md:
```diff
@@ -20,10 +20,15 @@ Then:
+to build the project. This command, as constructed in `esy.json`,
+
+1. generates a symbolically linked dir `bs-platform` in `node_modules`, which effectively makes `melange` the compiler behind the old `bsb` command, and then
+2. builds the project using the symbolically linked `bsb` command.
```
```suggestion
2. builds the project.
```

jchavarri reviewed (changes requested):
Thanks for this contribution! ✨ 

Please take a look at the requested changes, let me know what you think :)

Commit:
Apply suggestions from code review to to explain the linking to melange better

Co-authored-by: Javier Chávarri <javier.chavarri@gmail.com>

Commit:
Unignore `esy.lock`

Checking this folder in git makes sure everyone has the same versions of everything.

Co-authored-by: Javier Chávarri <javier.chavarri@gmail.com>

jchavarri reviewed (approved):
Thanks!
